### ISSUE TITLE
Reduce burden on users and maintainers via upgrade scripts

### ISSUE THREAD

**Issue by epage** [pksunkara/cargo-up#10]
Requiring users to pull and build large dependencies for one-off upgrades is a barrier to adoption.

What if `cargo-up` support rhai or lua upgrade scripts?  We only need to compile or install a pre-built binary of `cargo-up.  Crates could include the scripts in pre-determined locations in crates so they can be automatically be detected and ran.

**Comment by pksunkara** [comment]
The main reason for keeping this in rust is so that the migrators can use the underlying capability provided by [`Semantics`](https://docs.rs/cargo-up/0.0.5/cargo_up/type.Semantics.html) provided by `rust_analyzer` which can do a lot of cool stuff.

I am not sure about `rhai` and if we can do the same here.

And I guess for some easy migrators that don't need all that complex stuff, we can provide an alternative way to define the script.

> Requiring users to pull and build large dependencies for one-off upgrades is a barrier to adoption.

Definitely agree. Which is why we build the tool in `cargo-up-cache`. So it's only built for a long time during the first invocation on your system.

**Comment by epage** [comment]
Yeah, I fully agree that doing my suggestion would require writing a lot of bindings, the more featureful the scripts supported, the more bindings required.

**Comment by AllaithpPr** [comment]
S
